src/Models/SslCertificateCheck.php

```diff
@@ -75,6 +80,26 @@ public function isEnabled(): Attribute
         );
     }
 
+    public function shouldNotifyAboutFailure(): bool
+    {
+        $firstErrorTime = Cache::get('ssl_error_occurrence_time');
+
+        if ($firstErrorTime === null) {
+            return false;
+        }
+
+        $minutesSinceFirstFailure = now()->diffInMinutes($firstErrorTime);
+        $notificationInterval = config('moonguard.ssl_certificate_check.resend_invalid_certificate_notification_every_minutes');
+
+        if ($minutesSinceFirstFailure >= $notificationInterval) {
+            Cache::put('ssl_error_occurrence_time', now(), 60 * 24);
+
+            return true;
+        }
+
+        return false;
+    }
```

Comment:
This function should be inside `SslCertificateCheckService`:

Check out how it works on the UptimeCheckService:

This code should be inside the `SslCertificateCheckService`.

Check out how it works on UptimeCheckService: 

https://github.com/taecontrol/moonguard/blob/a840ca45b0fe4517d03de0ebfd2e55b31af2091e/src/Services/UptimeCheckService.php#L138-L153